camel-idea-plugin/src/main/java/org/apache/camel/idea/util/CamelIdeaUtils.java

```diff
@@ -126,12 +141,83 @@ public static boolean isCamelRouteSimpleExpression(PsiElement element) {
     }
 
     /**
+     * Is the given element a simple of a Camel route, eg <tt>simple</tt>, ot &lt;simple&gt;.
+     */
+    public static boolean isCameSimpleExpressionUsedAsPredicate(PsiElement element) {
+
+        // java
+        PsiMethodCallExpression call = PsiTreeUtil.getParentOfType(element, PsiMethodCallExpression.class);
```

Comment:
the element is newer a method call its a literal or something which has a simple text, eg "xxxx" but we need a fair bit of logic to find out which EIP called it, was it filter, transformer etc. Which can be done in different ways in Java DSL and hence all this logic below